spec/controllers/order_details_controller_spec.rb

```diff
@@ -346,4 +346,33 @@ def perform_update
       end
     end
   end
+
+  describe "#order_file" do
+    let(:product) { create(:setup_service, :with_order_form) }
+    let(:order_detail) { order.order_details.first }
+    let(:facility) { order.facility }
+
+    let(:params) { { order_id: order.id, id: order_detail.id } }
+    before { sign_in user }
+
+    describe "while in the cart" do
+      let(:order) { create(:setup_order, product: product) }
+
+      it "has access" do
+        get :order_file, params
+        expect(response).to be_success
+      end
+    end
+
+    describe "adding to an existing order" do
+      let(:order) { create(:purchased_order, product: product) }
+      let(:user) { create(:user, :staff, facility: facility) }
+      let(:merge_order) { create(:setup_order, product: product, account: order.account, user: order.user, merge_with_order_id: order.id) }
```

Comment:
More of a `TODO`, but merge order setup is complex enough to become a `trait` of `:setup_order`.